packages/story-editor/src/components/library/panes/media/common/insertionMenu.js

```diff
@@ -0,0 +1,197 @@
+import useFocusCanvas from '../../../../canvas/useFocusCanvas';
+
+const DropDownContainer = styled.div`
+  margin-top: 10px;
+  min-width: 160px;
+`;
+
+const MenuContainer = styled.div`
+  z-index: 1;
+`;
+
+const menuStylesOverride = css`
+  min-width: 160px;
+  margin-top: 0;
+  li {
+    display: block;
+  }
+`;
+
+// This is used for nested roving tab index to detect parent siblings.
+const BUTTON_NESTING_DEPTH = 3;
+const MENU_OPTIONS = {
+  INSERT: 'insert',
+  ADD_BACKGROUND: 'addBackground',
+};
+
+/**
+ * Get a More icon that displays a dropdown menu on click.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.resource Selected media element's resource object.
+ * @param {boolean} props.display Whether the more icon should be displayed.
+ * @param {Function} props.onInsert Callback for inserting media.
+ * @param {number} props.width Media width.
+ * @param {number} props.index Element index in the gallery.
+ * @param {boolean} props.isLocal If the menu is for local or 3p media.
+ * @return {null|*} Element or null if should not display the More icon.
+ */
+function InsertionMenu({ resource, display, onInsert, width, index, isLocal }) {
+  const insertButtonRef = useRef();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const onMenuOpen = useCallback((e) => {
+    e.stopPropagation();
+    setIsMenuOpen(true);
+  }, []);
+  const onMenuCancelled = useCallback(() => setIsMenuOpen(false), []);
+
+  const { currentBackgroundId, combineElements } = useStory((state) => ({
+    currentBackgroundId: state.state.currentPage?.elements?.[0]?.id,
+    combineElements: state.actions.combineElements,
+  }));
+
+  const { type, poster } = resource;
+  const insertLabel = ['image', 'gif'].includes(type)
+    ? __('Insert image', 'web-stories')
+    : __('Insert video', 'web-stories');
+  const options = [
+    {
+      group: [
+        { label: insertLabel, value: MENU_OPTIONS.INSERT },
+        {
+          label: __('Add as background', 'web-stories'),
+          value: MENU_OPTIONS.ADD_BACKGROUND,
+        },
+      ],
+    },
+  ];
+
+  const handleCurrentValue = (evt, value) => {
+    const thumbnailUrl = poster || getSmallestUrlForWidth(width, resource);
+    const newElement = getElementProperties(resource.type, {
+      resource,
+    });
+    onMenuCancelled();
+    switch (value) {
+      case MENU_OPTIONS.INSERT:
+        onInsert(resource, thumbnailUrl);
+        break;
+      case MENU_OPTIONS.ADD_BACKGROUND:
+        combineElements({
+          firstElement: newElement,
+          secondId: currentBackgroundId,
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  useRovingTabIndex({ ref: insertButtonRef }, [], BUTTON_NESTING_DEPTH);
+  // In case of 3p media, we tab out of the panel since it's the only button there.
+  const focusCanvas = useFocusCanvas();
+  useKeyDownEffect(isLocal ? null : insertButtonRef, 'tab', focusCanvas, [
```

Comment:
We could perhaps create a follow-up issue for it, seems like a general code quality question since that's currently indeed the way to do it.

Reply:
Yeah, I know. It just looks a bit ugly, and it's only because I know that you want to do a conditional hook, that I know what the code is supposed to do. If I didn't know that, this would confuse the hell out of me! 😂 

Follow-up is probably a good idea. I think I've done similar tricks elsewhere and felt bad about it.

Reply:
Added a follow-up issue: https://github.com/GoogleForCreators/web-stories-wp/issues/10670